test/test_data/test_feature_validator.py

```diff
@@ -0,0 +1,355 @@
+                                             return_X_y=True, as_frame=True)
+        return X
+    else:
+        ValueError("Unsupported indirect fixture {}".format(request.param))
+
+
+# Actual checks for the features
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'numpy_categoricalonly_nonan',
+        'numpy_numericalonly_nonan',
+        'numpy_mixed_nonan',
+        'numpy_categoricalonly_nan',
+        'numpy_numericalonly_nan',
+        'numpy_mixed_nan',
+        'pandas_categoricalonly_nonan',
+        'pandas_numericalonly_nonan',
+        'pandas_mixed_nonan',
+        'pandas_numericalonly_nan',
+        'list_numericalonly_nonan',
+        'list_numericalonly_nan',
+        'sparse',
+        'openml_40981',  # Australian
+    ),
+    indirect=True
+)
+def test_featurevalidator_supported_types(input_data_featuretest):
+    validator = FeatureValidator()
+    validator.fit(input_data_featuretest)
+    transformed_X = validator.transform(input_data_featuretest)
+    if sparse.issparse(input_data_featuretest):
+        assert sparse.issparse(transformed_X)
+    else:
+        assert isinstance(transformed_X, np.ndarray)
+    assert np.shape(input_data_featuretest) == np.shape(transformed_X)
+    assert np.issubdtype(transformed_X.dtype, np.number)
+    assert validator._is_fitted
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'list_categoricalonly_nonan',
+        'list_categoricalonly_nan',
+        'list_mixed_nonan',
+        'list_mixed_nan',
+    ),
+    indirect=True
+)
+def test_featurevalidator_unsupported_list(input_data_featuretest):
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r".*When providing a list of features.*Numpy failed"):
+        validator.fit(input_data_featuretest)
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'numpy_string_nonan',
+        'numpy_string_nan',
+    ),
+    indirect=True
+)
+def test_featurevalidator_unsupported_numpy(input_data_featuretest):
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r".*When providing a numpy array.*not supported."):
+        validator.fit(input_data_featuretest)
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'pandas_categoricalonly_nan',
+        'pandas_mixed_nan',
+        'openml_179',  # adult workclass has NaN in columns
+    ),
+    indirect=True
+)
+def test_featurevalidator_unsupported_pandas(input_data_featuretest):
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r"Categorical features in a dataframe.*missing/NaN"):
+        validator.fit(input_data_featuretest)
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'numpy_categoricalonly_nonan',
+        'numpy_mixed_nonan',
+        'numpy_categoricalonly_nan',
+        'numpy_mixed_nan',
+        'pandas_categoricalonly_nonan',
+        'pandas_mixed_nonan',
```

Comment:
How about adding checks for lists and sparse data?

Reply:
Done!